docs/dg/dev/architecture/coding-conventions.md

```diff
@@ -0,0 +1,1385 @@
+        <controller>template</controller>
+        <action>index</action>
+        <visible>1</visible>
+        <pages>
+          <configurable-bundle-template-create>
+            <label>Create Configurable Bundle Template</label>
+            <title>Create Configurable Bundle Template</title>
+            <bundle>configurable-bundle-gui</bundle>
+            <controller>template</controller>
+            <action>create</action>
+            <visible>0</visible>
+          </configurable-bundle-template-create>
+        </pages>
+      </configurable-bundle-templates>
+    </pages>
+  </product>
+</config>
+```
+
+## Permission Plugin
+
+**Description**
+
+Permission plugins are a way to put a scope on the usage of an application during a request lifecycle.
+
+**Conventions**
+
+- Permission plugins MUST implement `\Spryker\Shared\PermissionExtension\Dependency\Plugin\PermissionPluginInterface`.
+
+**Guidelines**
+- When using one of the traits in a model, use a protected constant string to reference the permission plugin that will be used in the model. As an annotation for this string you should add `@uses` annotation to the `PermissionPlugin::KEY` string that you will compare against (see in example below).
+- The `\Spryker\[Application]\Kernel\PermissionAwareTrait` trait allows the model in the corresponding application to check if a permission is granted to an application user.
+
+**Example**
+
+```php
+class QuotePermissionChecker implements QuotePermissionCheckerInterface
+{
+use PermissionAwareTrait;
+
+    /**
+     * @uses \Spryker\Client\SharedCart\Plugin\ReadSharedCartPermissionPlugin::KEY
+     */
+    protected const PERMISSION_PLUGIN_KEY_READ_SHARED_CART = 'ReadSharedCartPermissionPlugin';
+
+    public function checkQuoteReadPermission(QuoteTransfer $quoteTransfer): bool
+    {
+        ...
+      return $this->can(
+        static::PERMISSION_PLUGIN_KEY_READ_SHARED_CART,
+        $quoteTransfer->getCustomer()->getCompanyUserTransfer()->getIdCompanyUser(),
+        $quoteTransfer->getIdQuote(),
+      );
+    }
+```
+
+## Persistence Schema
+
+**Description**
+
+The schema file defines the module's tables and columns (see [Propel Documentation - Schema XML](#https://propelorm.org/documentation/reference/schema.html)).
+
+**Conventions**
+
+- Table name MUST follow the format `spy_[module]_[domain entity]` (eg: `spy_customer_address`, `spy_customer_address_book`).
+- Tables MUST have an integer ID primary key, following the format `id_[module]_[domain entity]` (eg: `id_customer_address`, `id_customer_address_book`).
+- Table definitions MUST include the `phpName` attribute.
+- Table foreign key MUST follow the format `fk_[remote entity]` (eg: `fk_customer_address`, `fk_customer_address_book`).
+- Table foreign key definitions MUST include the `phpName` attribute.
+- Tables with sensitive primary key (eg: `spy_order` ID gives information about submitted order count) MUST include UUID field for public access purpose.
```

Comment:
Shouldn't we simply say "Identifiers used for external communication MUST use a UUID"?